### ISSUE TITLE
Websockets Client: Show address with "ws:" in server info and support serverinfo

### ISSUE THREAD

**Issue by def-** [ddnet/ddnet#2847]
Allow adding websockets servers to favorites and loading their serverinfo over websockets instead of udp.

@QingGo Want to give it a try?

**Comment by QingGo** [comment]
Sure! I will try it after fixed the XP problem.

**Comment by Robyt3** [comment]
Should use `tw-0.6+ws://` for websockets instead of `ws://`. The server should also register itself with the mastersrv with `tw-0.6+ws://` (and `tw-0.7+ws://`) addresses if it has support for connecting via websockets.

**Comment by heinrich5991** [comment]
> Should use `tw-0.6+ws://` for websockets instead of `ws://`. The server should also register itself with the mastersrv with `tw-0.6+ws://` […] addresses if it has support for connecting via websockets.

Yes! Maybe as `ddnet-19+ws://`? And maybe drop everything older than that.

> `tw-0.7+ws://`

I think we shouldn't introduce that.

**Comment by heinrich5991** [comment]
I can do the masterserver part of this.

**Comment by Robyt3** [comment]
> > Should use `tw-0.6+ws://` for websockets instead of `ws://`. The server should also register itself with the mastersrv with `tw-0.6+ws://` […] addresses if it has support for connecting via websockets.
> 
> Yes! Maybe as `ddnet-19+ws://`? And maybe drop everything older than that.
> 
> > `tw-0.7+ws://`
> 
> I think we shouldn't introduce that.

How would you connect to a server via the 0.7 protocol and websockets otherwise? I don't really like to make connecting via websockets exclusive to the ddnet protocol, especially as connecting via the 0.7 protocol and websockets currently works, although I see the argument that there would likely not be any original 0.7 servers reachable via websockets in the foreseeable future but only DDNet-based servers via sixup-translation.

For the ddnet/0.6 protocol we can definitely go with `ddnet-19+ws://` instead of `tw-0.6+ws://` and probably assume there are no older clients and servers that actually support websockets.

**Comment by heinrich5991** [comment]
> How would you connect to a server via the 0.7 protocol and websockets otherwise? I don't really like to make connecting via websockets exclusive to the ddnet protocol, especially as connecting via the 0.7 protocol and websockets currently works, although I see the argument that there would likely not be any original 0.7 servers reachable via websockets in the foreseeable future but only DDNet-based servers via sixup-translation.

We maintain a 0.7 protocol extension for the ddnet client because there are few servers that only implement the 0.7 protocol and that's likely not going to change because they're unmaintained (or unmaintained and vanilla).

Websockets are something that only DDNet has implemented so far, so I think we shouldn't create another protocol (websockets + 0.7) that no one uses currently, and that we're going to have to support going forward, if people start implementing it.

**Comment by Robyt3** [comment]
> Websockets are something that only DDNet has implemented so far, so I think we shouldn't create another protocol (websockets + 0.7) that no one uses currently, and that we're going to have to support going forward, if people start implementing it.

Hmm, yeah, you are probably right.
